neuraxle/hyperparams/space.py

```diff
@@ -74,172 +74,128 @@
-                patched_arg = RecursiveDict(**{
-                    name: self._patch_arg(value)
-                    for name, value in args[0].items()
-                }, separator=arg.separator)
-            else:
-                patched_arg = arg
-            patched_args.append(patched_arg)
-
-        return patched_args
-
-    def _patch_arg(self, arg):
-        if hasattr(arg, 'dist') and isinstance(arg.dist, rv_generic):
-            if isinstance(arg.dist, rv_discrete):
-                return ScipyDiscreteDistributionWrapper(arg)
-            if isinstance(arg.dist, rv_continuous):
-                return ScipyContinuousDistributionWrapper(arg)
-        else:
-            return arg
-
-    def __getitem__(self, item: str = None):
-        value = self.get(item)
-        if isinstance(value, RecursiveDict) and len(value) == 0:
-            raise ValueError('{} not found in {}'.format(item, self.__class__.__name__))
-        return value
-
-    def get(self, item):
-        item_values = type(self)()
-        for name, values in self.items():
-            if item is None and not self.separator in name:
-                item_values[name] = values
-
-            if self.separator in name:
-                name_split = name.split(self.separator)
-                if str(name_split[0]) == item:
-                    item_values[self.separator.join(name_split[1:])] = values
-
-            if item == name:
-                item_values = values
-        return item_values
-
-    def to_flat(self) -> 'RecursiveDict':
-        """
-        Will create an equivalent flat HyperparameterSamples.
-
-        :return: an HyperparameterSamples like self, flattened.
-        """
-        return self.nested_dict_to_flat(dict_ctor=type(self))
-
-    def to_nested_dict(self) -> 'RecursiveDict':
-        """
-        Will create an equivalent nested dict HyperparameterSamples.
+            self.update(arg)
+        self.update(kwds)
+        self._patch_args()
 
-        :return: an HyperparameterSamples like self, as a nested dict.
-        """
-        return self.flat_to_nested_dict(dict_ctor=type(self))
+    def _is_only_arg_a_recursive_dict(self, args, kwds):
+        return len(args) == 1 and isinstance(args[0], RecursiveDict) and len(kwds) == 0
 
-    def to_flat_as_dict_primitive(self) -> dict:
-        """
-        Will create an equivalent flat HyperparameterSpace, as a dict.
+    def _patch_args(self):
+        to_patch_key_values = []
+        for k, v in self.items():
+            if isinstance(v, RecursiveDict):
+                v._patch_args()
+            else:
+                patched_arg, did_patch = self._patch_arg(v)
+                if did_patch:
+                    to_patch_key_values.append((k, patched_arg))
 
-        :return: an HyperparameterSpace like self, flattened.
-        """
-        return self.nested_dict_to_flat(dict_ctor=dict)
+        for k, v in to_patch_key_values:
+            self[k] = v
```

Comment:
same comment as above here for this loop. I didn't check really so I may be wrong.